src/net/sourceforge/kolmafia/StaticEntity.java

```diff
@@ -87,72 +90,104 @@ protected MafiaState initialValue()
 
 	public static final String getVersion()
 	{
-		return StaticEntity.getVersion( false );
-	}
-
-	public static final String getVersion( final boolean forceRevision )
-	{
-		String version = KoLConstants.VERSION_NAME;
-		if ( !KoLConstants.RELEASED || forceRevision )
+		if ( StaticEntity.cachedversionName == null )
 		{
-			int revision = StaticEntity.getRevision();
-			if ( revision != 0 )
-			{
-				version += " r" + revision;
-			}
+			StaticEntity.cachedversionName = PRODUCT_NAME + " r" + StaticEntity.getRevision();
 		}
-		return version;
+
+		return StaticEntity.cachedversionName;
 	}
 
 	public static final int getRevision()
 	{
+		if ( StaticEntity.cachedRevisionNumber != null )
+		{
+			return StaticEntity.cachedRevisionNumber;
+		}
+
+		// Get the revision from the jar manifest attributes
 		try
 		{
-			if ( StaticEntity.cachedSVNRevisionNumber != null )
-			{
-				return StaticEntity.cachedSVNRevisionNumber;
-			}
-			if ( KoLConstants.REVISION == null && SVNWCUtil.isWorkingCopyRoot( KoLConstants.ROOT_LOCATION ) )
+			ClassLoader classLoader = StaticEntity.class.getClassLoader();
+			if ( classLoader != null )
 			{
-				SVNInfo info = SVNManager.doInfo( KoLConstants.ROOT_LOCATION );
-				StaticEntity.cachedSVNRevisionNumber = (int) info.getRevision().getNumber();
-				return StaticEntity.cachedSVNRevisionNumber;
+				URL resources = classLoader.getResource( "META-INF/MANIFEST.MF" );
+
+				Manifest manifest = new Manifest( resources.openStream() );
+				String buildRevision = manifest.getMainAttributes().getValue( "Build-Revision" );
+				if ( buildRevision != null && StringUtilities.isNumeric( buildRevision ) )
```

Comment:
Optional: As noted in the previous PR, I think wrapping Integer.parseInt in a try/catch would be sufficient here and not be subject to the nonstandard parsing rules present in StringUtilities.isNumeric / parseInt.

Reply:
(And as we both noted, it's not a meaningful change of behavior, in most cases. It just accepts and caches a slightly broader set of inputs.)